code/modules/mob/living/silicon/ai/freelook/chunk.dm

```diff
@@ -21,26 +21,41 @@
 
 // Add an AI eye to the chunk, then update if changed.
 
-/datum/camerachunk/proc/add(mob/camera/aiEye/ai)
-	if(!ai.ai)
-		return
-	ai.visibleCameraChunks += src
-	if(ai.ai.client)
-		ai.ai.client.images += obscured
+/datum/camerachunk/proc/add(mob/camera/aiEye/eye)
+	if(istype(eye,/mob/camera/aiEye/remote))

```

Comment:
There are more instances of this bad pattern in this PR diff. This will make the code less maintainable and harder to implement more features in the future.